mysite/announcements/models.py

```diff
@@ -32,3 +31,24 @@ def clean(self):
             raise ValidationError("The date and time can't be in past")
         if self.date_time_expire <= self.date_time_to_publish:
             raise ValidationError("Scheduled time can't be less than expiry time")
+
+
+STATUS_CHOICES = (
+    ('seen', 'SEEN'),
+    ('yet to receive', 'YET TO RECEIVE'),
+    ('expired', 'EXPIRED'),
+    ('user left group', 'USER LEFT GROUP')
+)
+
+
+@python_2_unicode_compatible
+class Status(models.Model):
```

Comment:
Status is a very generic name, it does not explicitly define the status of what?

Status of Announcement? Status of Group? Status of Weather? Status of relationship?

Reply:
I would name it as Announcement delivery status